src/addrman.h

```diff
@@ -525,19 +527,26 @@ friend class CAddrManTest;
         if (format >= Format::V2_ASMAP) {
             s >> serialized_asmap_checksum;
         }
+        const bool restore_bucketing{nUBuckets == ADDRMAN_NEW_BUCKET_COUNT &&
+                                     serialized_asmap_checksum == supplied_asmap_checksum};
 
         for (auto bucket_entry : bucket_entries) {
             int bucket{bucket_entry.first};
             const int entry_index{bucket_entry.second};
             CAddrInfo& info = mapInfo[entry_index];
+
+            // The entry shouldn't appear in more than
+            // ADDRMAN_NEW_BUCKETS_PER_ADDRESS. If it has already, just skip
+            // this bucket_entry.
+            if (info.nRefCount >= ADDRMAN_NEW_BUCKETS_PER_ADDRESS) continue;
+
             int nUBucketPos = info.GetBucketPosition(nKey, true, bucket);
-            if (format >= Format::V2_ASMAP && nUBuckets == ADDRMAN_NEW_BUCKET_COUNT && vvNew[bucket][nUBucketPos] == -1 &&
```

Comment:
So you're removing the `format >= Format::V2_ASMAP` condition here because otherwise it'd be impossible for `serialized_asmap_checksum == supplied_asmap_checksum` yes?

Reply:
Correct. If format == V1_DETERMINISTIC, then serialized_asmap_checksum must be 0.